include/cell/copy/global_to_register.hpp

```diff
@@ -1,83 +1,190 @@
+                                                    tl::Layout::RowMajor>;
+                Loader loader;
+                loader(data, dst(i, j));
             }
         }
     }
 };
 
-/// @brief Load tile from global memory to register tile in ColMajor layout.
-/// @tparam Global Global tile type.
-/// @tparam Reg Register tile type.
-// TODO(KuangjuX): Implement LoadImpl for ColMajor layout.
-template <typename Global, typename Reg>
-struct GlobalToRegLoaderImpl<Global, Reg, tl::Layout::ColMajor> {};
+// TODO(KuangjuX): Implement this.
+template <typename Global_, typename Reg_, const int kRowExec_,
+          const int kColExec_>
+struct GlobalToRegLoaderImpl<Global_, Reg_, kRowExec_, kColExec_,
+                             tl::Layout::ColMajor> {
+    using Global = Global_;
+    using Reg = Reg_;
+    using DType = typename Global::DType;
+    using BaseTile = typename Reg::DType;
+
+    // strides to iterate over each 16x128-bits `BaseTile`.
+    static constexpr int kTileRstride =
+        BaseTileShape<DType>::row * Global::kRowStride;
+    static constexpr int kTileCstride = BaseTileShape<DType>::col;
+
+    // row stride and col stride for a thread in a warp
+    static constexpr int kStride = traits::TraitsBase<DType>::kNumPerAccess;
+    static constexpr int kLaneRstride = Global::kRowStride;
+    static constexpr int kLaneCstride = kStride;
 
-}  // namespace details
+    static constexpr int kRowExec = kRowExec_;
+    static constexpr int kColExec = kColExec_;
+
+    DEVICE void operator()(const DType* src, Reg& dst) {}
+};
 
-template <typename Global_, typename Reg_, tl::Layout type_>
-struct GlobalToRegLoader {
+/**
+ * @brief Load memory from Global memory to Register based on the warp reuse
+ * mode.
+ * @tparam Global_ Global memory tile type.
+ * @tparam Reg_ Register tile type.
+ * @tparam WarpLayout_ Warp layout type.
+ * @tparam kMode_ Warp reuse mode.
+ * @tparam Base Copy base.
+ */
+template <typename Global_, typename Reg_, typename WarpLayout_,
+          const WarpReuse kMode_,
+          typename Base = warp::CopyBase<WarpLayout_, kMode_>>
+struct GlobalToRegLoader : public Base {
     using Global = Global_;
     using Reg = Reg_;
     using DType = typename Global::DType;
+    using BaseTile = typename Reg::DType;
+
+    using WarpLayout = WarpLayout_;
+    static constexpr WarpReuse kMode = kMode_;
 
     DEVICE void operator()(const Global& src, Reg& dst) {
-        details::GlobalToRegLoaderImpl<Global, Reg, type_> loader;
-        loader(src, dst);
+        const DType* src_ptr = src.data();
+
+        // 1. advance the pointer to input data to the current warp
+        // according to warp reuse mode.
+        src_ptr += Base::template get_warp_offset<Global>();
+
+        // how many times a `BaseTile` is executed along the row and column
+        // direction.
+        static constexpr int kRowExec =
+            Base::template row_exec_count<BaseTile, Global::kRows>();
```

Comment:
I believe `BaseTile` (also on line 182) should be `BaseTileShape<DType>`. The `16x16` tile shape should be passed to `xx_exec_count` instead of the `BaseTile` for a single thread.

Reply:
fix.